nwb_conversion_tools/utils/spike_interface.py

```diff
@@ -331,27 +332,65 @@ def add_electrodes(
     )
     if metadata is None:
         metadata = dict(Ecephys=dict())
+
+    if 'Ecephys' not in metadata:
+        metadata['Ecephys'] = dict()
+
     if 'Electrodes' not in metadata['Ecephys']:
         metadata['Ecephys']['Electrodes'] = []
 
     assert all([isinstance(x, dict) and set(x.keys()) == set(['name', 'description', 'data'])
                 and isinstance(x['data'], list) for x in metadata['Ecephys']['Electrodes']]), \
         "Expected metadata['Ecephys']['Electrodes'] to be a list of dictionaries!"
     assert all([x['name'] != 'group' for x in metadata['Ecephys']['Electrodes']]), \
-        "Passing metadata field 'group' is depricated; pass group_name instead!"
+        "Passing metadata field 'group' is deprecated; pass group_name instead!"
 
     if nwbfile.electrodes is None:
         nwb_elec_ids = []
     else:
         nwb_elec_ids = nwbfile.electrodes.id.data[:]
 
-    for metadata_column in metadata['Ecephys']['Electrodes']:
-        if (nwbfile.electrodes is None or metadata_column['name'] not in nwbfile.electrodes.colnames) \
-                and metadata_column['name'] not in defaults:
-            nwbfile.add_electrode_column(
-                name=str(metadata_column['name']),
-                description=str(metadata_column['description'])
-            )
+    elec_columns = {}  # name: description
+    property_names = set()
+    for chan_id in recording.get_channel_ids():
+        for i in recording.get_channel_property_names(channel_id=chan_id):
+            property_names.add(i)
+    for prop in property_names:
+        if prop not in ['gain', 'offset', 'location', 'name'] + list(exclude):
+            # property 'gain' should not be in the NWB electrodes_table
+            # property 'brain_area' of RX channels corresponds to 'location' of NWB electrodes
+            # property 'offset' should not be in the NWB electrodes_table as not officially supported by schema v2.2.5
+
+            data = []
+            for chan_id in recording.get_channel_ids():
+                if prop in recording.get_channel_property_names(channel_id=chan_id):
+                    data.append(recording.get_channel_property(channel_id=chan_id, property_name=prop))
+                else:
+                    if len(data) > 0 and isinstance(data[-1], str):
+                        data.append('')
+                    elif len(data) > 0:
+                        data.append(np.nan)
```

Comment:
I see that you are trying to fill missing data with auto-detected formats to keep the entire data vector the length of the channel ids, but there is an issue here that could occur if a property exists for any channel after the first one, since the data will not have been appended at least once (the `len `will still be zero) meaning no auto-fill occurred prior to observing that property.

I understand and appreciate this approach as it differs from relying on `get_shared_channel_properties`, and I definitely think there is a place for it, but it has to be thought of very carefully.

Reply:
yes, thats a case where it could fail, can you review [these changes](https://github.com/catalystneuro/nwb-conversion-tools/pull/181/commits/70e13898d4ccd111fe3c7e9dc9e39a73ce52f434)